test/invariants.md

```diff
@@ -0,0 +1,79 @@
+- An open policy can be applied to any calling contract by the calling contract admin 
+- A disabled policy will never be envoked within the checkPolicies function 
+- Only the policy admin for the policy can add a subscriber 
+- Only the policy admin for the policy can update a policy 
+- Only the policy admin for the policy can delete a policy 
+- Only the policy admin can close the policy 
+- Only the policy admin can open the policy 
+- Only the policy admin can cement the policy 
+- Only the policy admin can disable the policy 
+- Only the policy admin can enable the policy once disabled
+- Only the calling contract admin can apply the policy to ther contract
+- Only the calling contract admin can un apply the policy from ther contract
+
+# Rules Invariants 
+- Only the policy admin for the policy can create a rule within the policy 
+- Only the policy admin for the policy can delete a rule within the policy 
+- Only the policy admin for the policy can update a rule within the policy 
+- RuleId can only ever increment within a policy 
+
+
+# Components Invariants 
+## Foreign Call Invariants 
+- Only the policy admin for the policy can create a foreign call within the policy 
+- Only the policy admin for the policy can delete a foreign call within the policy 
+- Only the policy admin for the policy can update a foreign call within the policy
+- foreignCallIdxCounter can only ever inrement within a policy
+
+## Calling Function Invariants 
+- Only the policy admin for the policy can create a calling function within the policy 
+- Only the policy admin for the policy can delete a calling function within the policy 
+- Only the policy admin for the policy can update a calling function within the policy
+- Deleting a calling function will always delete associated rules 
+- callingFunctionId can only ever inrement within a policy
+
+# Effect Invariants 
+- When rule conditionals evaluate to true, the positive effect(s) will always execute 
+- When rule conditionals evaluate to false, the negative effect(s) will always execute 
+- When a revert effect is executed, all transaction changes are reverted and no changes to state are stored. 
+- The reversion message will always match the effect errorMessage for the executing rule. 
+
+# Tracker Invariants 
+- Only the policy admin for the policy can create a tracker within the policy 
+- Only the policy admin for the policy can delete a tracker within the policy 
+- Only the policy admin for the policy can update a tracker within the policy
+- trackerIdxCounter can only ever inrement within a policy
+
+# Diamond Pattern Invariants 
+- Diamond Facets Cannot be upgraded once the diamond is initialized 
+- Diamond Facets Cannot be deleted once the diamond is initialized  
+- Diamond Facets Cannot be replaced once the diamond is initialized
+- Diamond can only be initialized once 
+- Diamond Ownership will be transferred to the address that deploys the Rules Engine  
```

Comment:
Do we want this? Doesn't this mean that the diamond can upgrade? Why wouldn't we just push it to the burn address?